virtualization-and-containers/material/virtualmachines_slides.md

```diff
@@ -0,0 +1,116 @@
+  }
+  .reveal p {
+    text-align: left;
+  }
+  .reveal section h1 {
+    color: orange;
+  }
+  .reveal section h2 {
+    color: orange;
+  }
+</style>
+
+# Virtual Machines
+
+---
+
+## What is a Virtual Machine?
+
+<img src="https://raw.githubusercontent.com/Simulation-Software-Engineering/Lecture-Material/main/virtualization-and-containers/material/figs/virtualmachine-sketch.png" width=60%; style="margin-left:auto; margin-right:auto; padding-top: 25px; padding-bottom: 25px">
+
+- Environment that allows you to run any (supported) operating
+- Virtual Machines are portable and flexible
+
+---
+
+## Common Terms
+
+- Host operating system (host OS)
+  - The OS the hypervisor is installed on
+- Guest operating system (guest OS)
+  - The OS running inside the virtual machine
+- Virtual machine (VM)
+  - Environment the guest is running in
+
+---
+
+## Types of Virtual Machines / Hypervisors
+
+- Sometimes distinction not clear, but in general:
+  - **Type 1**
+    - Runs directly on bare-metal hardware
+    - Examples: [Microsoft Hyper-V](https://docs.microsoft.com/en-us/virtualization/hyper-v-on-windows/about/), [VMware ESXi](https://www.vmware.com/products/esxi-and-esx.html), [Proxmox Virtual Environment](https://www.proxmox.com/en/proxmox-ve), [Xen](https://xenproject.org/)...
+  - **Type 2**
+    - Requires a running OS
+    - Examples: [VirtualBox](https://www.virtualbox.org/), [VMWare Workstation Player](https://www.vmware.com/products/workstation-player.html), [Parallels (for Mac)](https://www.parallels.com/eu/products/desktop/)...
+
+---
+
+## (Some) Features of Virtual Machines
+
+- VM might obtain exclusive access to some of your resources
+- Behaves (more or less) as native installation (root etc., system calls)
+- (Strict) separation from host operating system
+- Popular for
+  - Safety critical tasks
+  - Development and testing
+  - Wherever one wants a portable solution
+
+---
+
+## Why is Virtualization Useful?
+
+- Running multiple operating systems simultaneously
+  - Test/develop software for other OS, debug problems on other OS, Windows on Linux, fill in forms of university (MS Office)
+- Easier software installations and testing
+  - Preconfigured VMs trainings and teaching
+- Testing and disaster recovery
+  - Create snapshots beforetesting, copy VMs etc.
+- Infrastructure consolidation
+  - Run many VMs on single host
+- Seperate services from each other (security?!)
+
+---
+
+## Short Note on Security
+
+- Does **not** solve all problems
+- Separation from Host, but VM also tightly embedded in Host
+- Virtual machine / Hypervisor might increase attack surface
+- Additional layer of complexity
+- Read the manual
+  - Examples: GPU acceleration, networking, USB passthrough...
+- Think about **what** you do and **before** you do it
+
+---
+
+## Summary
+
+- Encapsulated environment that emulates a computer
+- Type 1 and 2 Hypervisors
+- Run additional OSes on your machine
+- Portable VMs
+- Keep security in mind
```

Comment:
Maybe this style of writing is a bit short. Sth like "VM are ...", "There is type 1 and ..." could be helpful